js/ai.js

```diff
@@ -67,6 +67,14 @@ export default {
         this.test = 0;
         this.foundSet = false;
 
+        //have ai check each time it resumes if game should end; copied from board logic
+        if (isDev)  console.log(`ai sets found: ${this.findValidSets().length}`);
```

Comment:
bring out the computation of findvalidsets into a call like let numValidSets = ....length; to avoid double computation (not needed for stock because it's a trivial lookup)

Reply:
Done in latest commit